src/backendMiddleware.ts

```diff
@@ -40,4 +66,120 @@ export class BackendMiddleware {
   public async initStorage(): Promise<void> {
     await this.storageLib.initConnection()
   }
+
+  public get identityWallet(): IdentityWallet {
+    if (this._identityWallet) return this._identityWallet
+    throw new BackendError(ErrorCodes.NoWallet)
+  }
+
+  public get keyProvider(): SoftwareKeyProvider {
+    if (this._keyProvider) return this._keyProvider
+    throw new BackendError(ErrorCodes.NoKeyProvider)
+  }
+
+  public async prepareIdentityWallet(): Promise<IdentityWallet> {
+    if (this._identityWallet) return this._identityWallet
+
+    const encryptedEntropy = await this.storageLib.get.encryptedSeed()
+    if (!encryptedEntropy) throw new BackendError(ErrorCodes.NoEntropy)
+    const encryptionPass = await this.keyChainLib.getPassword()
+
+    const decryptedSeed = this.encryptionLib.decryptWithPass({
+      cipher: encryptedEntropy,
+      pass: encryptionPass,
+    })
+
+    if (!decryptedSeed) {
+      throw new BackendError(ErrorCodes.DecryptionFailed)
+    }
+
+    // TODO: rework the seed param on lib, currently cleartext seed is being passed around. Bad.
+    this._keyProvider = JolocomLib.KeyProvider.fromSeed(
+      Buffer.from(decryptedSeed, 'hex'),
+      encryptionPass,
+    )
+
+    const userPubKey = this._keyProvider.getPublicKey({
+      derivationPath: KeyTypes.jolocomIdentityKey,
+      encryptionPass,
+    })
+
+    const didDocument = await this.storageLib.get.didDoc(
+      publicKeyToDID(userPubKey),
```

Comment:
We are accessing the DID Document based on the DID which is based on the public key, which will not work, once we have key rotations.

Reply:
Yes true, but this is PR is only meant to be a refactor.

Reply:
Yes true, but this is PR is only meant to be a refactor.

Reply:
This is now in #1427